pkg/ccl/backupccl/restore_span_covering.go

```diff
@@ -182,45 +200,72 @@ func makeSimpleImportSpans(
 						break
 					}
 					f := it.Value()
-					fspan := endKeyInclusiveSpan(f.Span)
-					if sp := span.Intersect(fspan); sp.Valid() {
-						fileSpec := execinfrapb.RestoreFileSpec{Path: f.Path, Dir: backups[layer].Dir}
-						if dir, ok := backupLocalityMap[layer][f.LocalityKV]; ok {
-							fileSpec = execinfrapb.RestoreFileSpec{Path: f.Path, Dir: dir}
-						}
+					fileSpec := execinfrapb.RestoreFileSpec{Path: f.Path, Dir: backups[layer].Dir}
+					if dir, ok := backupLocalityMap[layer][f.LocalityKV]; ok {
+						fileSpec = execinfrapb.RestoreFileSpec{Path: f.Path, Dir: dir}
+					}
 
-						// Lookup the size of the file being added; if the backup didn't
-						// record a file size, just assume it is 16mb for estimating.
-						sz := f.EntryCounts.DataSize
-						if sz == 0 {
-							sz = 16 << 20
+					// Lookup the size of the file being added; if the backup didn't
+					// record a file size, just assume it is 16mb for estimating.
+					sz := f.EntryCounts.DataSize
+					if sz == 0 {
+						sz = 16 << 20
+					}
+
+					if inclusiveOverlap(span, f.Span) {
+						var startKey roachpb.Key
+						var endKey roachpb.Key
+						if sp := span.Intersect(f.Span); sp.Valid() {
+							startKey = sp.Key
+							endKey = sp.EndKey
+						} else {
+							startKey = f.Span.EndKey
+							endKey = f.Span.EndKey
 						}
 
 						if len(cover) == spanCoverStart {
-							cover = append(cover, makeEntry(span.Key, sp.EndKey, fileSpec))
-							lastCovSpanSize = sz
+							entry := makeEntry(span.Key, endKey)
+							if entry.Span.Valid() {
+								lastCovSpanSize = endKeyNotCoveredFiles.drain(&entry)
```

Comment:
I think this might be slightly easier to read either as `lastCovSpanSize = 0; lastCovSpanSize += drain() ... lastCovSpanSize+= sz` so the reset it clear and then both components use `+=`, or as `drainedSize := drain(); ... lastCovSpanSize = drainedSz + sz` so it is re-assigned just once.

Reply:
It might also be nice to comment above here explaining why we can assume everything endKeyNotCoveredFiles is safe to drain into this [span.Key, endKey) entry, i.e. it covered anything undrained

Reply:
could the "make a new span" logic get pulled into its own function? It may be easier on the eyes. I see this block of code in two places:
```
entry := makeEntry(covEnd, endKey)
lastCovSpanSize = endKeyNotCoveredFiles.drain(&entry)

entry.Files = append(entry.Files, fileSpec)
lastCovSpanSize += sz

cover = append(cover, entry)
```

It doesn't seem that you need to make the Entry outside the branch here.
```
entry := makeEntry(span.Key, endKey)
if entry.Span.Valid() 
```

Reply:
Done, I explicitly added `lastCovSpanSize = 0` to make it clear that the size is being reset. 

> could the "make a new span" logic get pulled into its own function?

I think that given this is depreciating code I'd rather just keep all of the logic inline for now.